independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,250 @@
+                .filter(dep -> dep.isResolved() &&
+                        dep.isFlagSet(TOP_LEVEL_RUNTIME_EXTENSION_ARTIFACT) &&
+                        onlyPublic(dep.getGroupId(), log))
+                .map(dep -> TrackProperties.AppExtension.builder()
+                        .groupId(dep.getGroupId())
+                        .artifactId(dep.getArtifactId())
+                        .version(dep.getVersion())
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    void sendIdentity(final Map<String, Object> context) {
+        if (this.userId.isNew()) { // fixme when inactive on 1st call it will not send identity.
+            this.restClient.postIdentity(Identity.builder()
+                    .userId(this.userId.getUuid())
+                    .context(context)
+                    .timestamp(Instant.now())
+                    .build());
+        }
+    }
+
+    Map<String, Object> createContextMap(ApplicationModel applicationModel,
+            Map<String, Object> buildInfo) {
+        WorkspaceModuleId moduleId = applicationModel.getApplicationModule().getId();
+
+        return new ContextBuilder()
+                .mapPair(PROP_APP)
+                .pair(PROP_NAME, hashSHA256(moduleId.getGroupId() + ":" + moduleId.getArtifactId()))
+                .pair(PROP_VERSION, hashSHA256(moduleId.getArtifactId() + ":" + moduleId.getVersion()))
+                .build()
+                .mapPair(PROP_JAVA)
+                .pair(PROP_VENDOR, getProperty("java.vendor", "N/A"))
+                .pair(PROP_VERSION, getProperty("java.version", "N/A"))
+                .build()
+                .mapPair(PROP_GRAALVM)
+                .pair(PROP_VENDOR, ofNullable(buildInfo.get(GRAALVM_VERSION_DISTRIBUTION)).orElse("N/A"))
+                .pair(PROP_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_VERSION)).orElse("N/A"))
+                .pair(PROP_JAVA_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_JAVA)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_BUILD)
+                .pair(PROP_MAVEN_VERSION, ofNullable(buildInfo.get(MAVEN_VERSION)).orElse("N/A"))
+                .pair(PROP_GRADLE_VERSION, ofNullable(buildInfo.get(GRADLE_VERSION)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_QUARKUS)
+                .pair(PROP_VERSION, getQuarkusVersion(applicationModel))
+                .build()
+                .pair(PROP_IP, VALUE_NULL_IP)
+                .mapPair(PROP_LOCATION)
+                .pair(PROP_LOCALE_COUNTRY, Locale.getDefault().getCountry())
+                .build()
+                .mapPair(PROP_OS)
+                .pair(PROP_NAME, getProperty("os.name", "N/A"))
+                .pair(PROP_VERSION, getProperty("os.version", "N/A"))
+                .pair(PROP_OS_ARCH, getProperty("os.arch", "N/A"))
+                .build()
+                .mapPair(PROP_CI)
+                .pair(PROP_CI_NAME, getBuildSystemName())
```

Comment:
Does it make sense to have a property that tells whether the `CI` environment is set to indicate that it's running in CI (in case `getBuildSystemName` returns `unknown` for some exotic CI)?

Reply:
There are a limited number of CIs we can detect. It returns unknown when he couldn't detect any CI.